File: server/makc2022--nodejs--nestjs/src/layer1/query/query-result-with-output.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { QueryResult } from './query-result';

/** Результат запроса с выходными данными.
 * @template TOutput Тип выходных данных.
 */
export class QueryResultWithOutput<TOutput> extends QueryResult {
  /** Выходные данные. */
  output: TOutput;
}

File: server/makc2022--dotnet/Makc2022.Layer1/Query/QueryResultWithOutput.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Makc2022.Layer1.Query
{
    /// <summary>
    /// Результат запроса с выходными данными.
    /// </summary>
    /// <typeparam name="TOutput">Тип выходных данных.</typeparam>
    public class QueryResultWithOutput<TOutput> : QueryResult
    {
        #region Properties

        /// <summary>
        /// Выходные данные.
        /// </summary>
        public TOutput? Output { get; set; }

        #endregion Properties
    }
}
